File: macos/Scanner/Sources/Scanner/Detectors/VSCodeDetector.swift
public struct VSCodeDetector: Detector {
    public let id = "vscode"
    public let displayName = "VS Code"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: ["/Applications/Visual Studio Code.app", homePath(fs, "Applications/Visual Studio Code.app")]
        )

        let extensionsPath = homePath(fs, ".vscode/extensions")
        if fs.directoryExists(extensionsPath) {
            facts.extensions.append(contentsOf: try parseExtensions(fs.listDirectoryNames(extensionsPath)))
        }

        let continuePath = homePath(fs, ".continue/config.json")
        if let text = try readIfPresent(fs, path: continuePath) {
            facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: continuePath))
        }

        let storageRoot = homePath(fs, "Library/Application Support/Code/User/globalStorage")
        for extensionDir in ["saoudrizwan.claude-dev", "rooveterinaryinc.roo-cline"] {
            let root = storageRoot + "/" + extensionDir
            guard fs.directoryExists(root) else {
                continue
            }
            let candidates = try fs.listFilesRecursively(root, maxDepth: 3)
                .filter { path in
                    let lower = path.lowercased()
                    return lower.hasSuffix(".json") && lower.contains("mcp")
                }
            for path in candidates {
                if let text = try readIfPresent(fs, path: path) {
                    facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: path))
                }
            }
        }

        return facts
    }

    private func parseExtensions(_ folders: [String]) -> [ExtensionFact] {
        folders.compactMap { folder in
            if folder.hasPrefix("saoudrizwan.claude-dev-") {
                return ExtensionFact(appId: id, extensionId: "saoudrizwan.claude-dev", extensionName: "Cline", hasTerminalAccess: true)
            }
            if folder.hasPrefix("rooveterinaryinc.roo-cline-") {
                return ExtensionFact(appId: id, extensionId: "rooveterinaryinc.roo-cline", extensionName: "Roo", hasTerminalAccess: true)
            }
            if folder.hasPrefix("continue.continue-") {
                return ExtensionFact(appId: id, extensionId: "continue.continue", extensionName: "Continue", hasTerminalAccess: true)
            }
            if folder.hasPrefix("github.copilot-") {
                return ExtensionFact(appId: id, extensionId: "github.copilot", extensionName: "GitHub Copilot", hasTerminalAccess: false)
            }
            return nil
        }
    }
}

File: windows/Scanner/Detectors/VSCodeDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class VSCodeDetector : IDetector
{
    public string Id => "vscode";
    public string DisplayName => "VS Code";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [Path.Combine(fs.LocalApplicationDataDirectory, "Programs", "Microsoft VS Code")]
        );

        var extensionsPath = DetectorSupport.HomePath(fs, ".vscode/extensions");
        if (fs.DirectoryExists(extensionsPath))
        {
            facts.Extensions.AddRange(ParseExtensions(fs.ListDirectoryNames(extensionsPath)));
        }

        var continuePath = DetectorSupport.HomePath(fs, ".continue/config.json");
        var continueText = DetectorSupport.ReadIfPresent(fs, continuePath);
        if (continueText is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(continueText, Id, continuePath));
        }

        var storageRoot = Path.Combine(fs.ApplicationDataDirectory, "Code", "User", "globalStorage");
        foreach (var extensionDir in new[] { "saoudrizwan.claude-dev", "rooveterinaryinc.roo-cline" })
        {
            var root = Path.Combine(storageRoot, extensionDir);
            if (!fs.DirectoryExists(root))
            {
                continue;
            }

            foreach (var path in fs.ListFilesRecursively(root, 3).Where(IsMcpJsonFile))
            {
                var text = DetectorSupport.ReadIfPresent(fs, path);
                if (text is not null)
                {
                    facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(text, Id, path));
                }
            }
        }

        return facts;
    }

    private IEnumerable<ExtensionFact> ParseExtensions(IEnumerable<string> folders)
    {
        foreach (var folder in folders)
        {
            if (folder.StartsWith("saoudrizwan.claude-dev-", StringComparison.Ordinal))
            {
                yield return new ExtensionFact(Id, "saoudrizwan.claude-dev", "Cline", true);
            }
            else if (folder.StartsWith("rooveterinaryinc.roo-cline-", StringComparison.Ordinal))
            {
                yield return new ExtensionFact(Id, "rooveterinaryinc.roo-cline", "Roo", true);
            }
            else if (folder.StartsWith("continue.continue-", StringComparison.Ordinal))
            {
                yield return new ExtensionFact(Id, "continue.continue", "Continue", true);
            }
            else if (folder.StartsWith("github.copilot-", StringComparison.Ordinal))
            {
                yield return new ExtensionFact(Id, "github.copilot", "GitHub Copilot", false);
            }
        }
    }

    private static bool IsMcpJsonFile(string path)
    {
        var lower = path.ToLowerInvariant();
        return lower.EndsWith(".json", StringComparison.Ordinal) && lower.Contains("mcp", StringComparison.Ordinal);
    }
}
